### PR TITLE
Add pod antiaffinity to kubedns template

### PR DESCRIPTION
**Pull request by galal-hussein**
#302

### CODE REVIEW

File: templates/kubedns.go [hunk 1]
```diff
@@ -113,6 +113,17 @@ spec:
       annotations:
         scheduler.alpha.kubernetes.io/critical-pod: ''
     spec:
+      affinity:
```
Comment (alena1108) [inline templates/kubedns.go]: I recall seeing k8s issue in 1.6 for upgrades. With limited number of hosts, and self anti affinity, it was possible for the upgrade to fail. Can you check with k8s upstream to see if the issue is still there. If it is, we might need to put some comment to this code, so next time on the addon upgrade, we would change the corresponding label, so the addons scheduling won'g get conflicts during the upgrade.

Comment (willmao) [reply templates/kubedns.go]: @alena1108 @galal-hussein 

why schedule kube-dns on the same node,? would it be a SPOF if the node has network issue ?

Comment (alena1108) [reply templates/kubedns.go]: @willmao it is `anti` affinity rule. Implying the pods won't be on the same node

Comment (willmao) [reply templates/kubedns.go]: sorry, i misunderstood the setting, but one day i saw the kube-dns pods scheduled on the same node, so i found the code here..

thanks for your time!

**Pull request merged**